Assignment/ZooManagement/ZooManagement.java

```diff
@@ -0,0 +1,120 @@
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class will perform various operations on Zoo management
+ * @author Rachna Jadam
+ */
+public class ZooManagement {
+    List<Zone> listOfZone = new ArrayList<Zone>(); // create the list of zones
+	
+    /**
+     * This method add the zone for a specific type of animal
+     * @param zoneType
+     * @param capacity
+     * @param hasPark
+     * @param hasCanteen
+     */
+    public boolean addZone(AnimalFamily zoneType, int capacity, boolean hasPark, boolean hasCanteen)
+    {
+        Zone zone = null;		
+        switch(zoneType)
+        {
+            case BIRD:
+                zone = new BirdZone(capacity, hasPark, hasCanteen);
+                break;
+            case MAMMAL:
+                zone = new MammalZone(capacity, hasPark, hasCanteen);
+                break;
+            case REPTILE:
+                zone = new ReptileZone(capacity, hasPark, hasCanteen);
+                break;		
+        }
+        listOfZone.add(zone);
+        return true;
+    }
+	
+    /**
+     * This method will add cage to a particular zone
+     * @param animal      animal name for what you want to add cage
+     * @param capacity    capacity of cage for a particular animal
+     * @return            true if cage add successfully
+     * @exception         assertion error if no zone or animal present of a particular type
+     */
+    public boolean addCage(AnimalName animal, int capacity) throws AssertionError{
+        Cage cage = new Cage(animal, capacity);
```

Comment:
it should be written in such a way if new animal comes it will check for the cage.